picocli-shell-jline3/src/main/java/picocli/shell/jline3/PicocliCommands.java

```diff
@@ -47,14 +74,25 @@ public PicocliCommands(Path workDir, CommandLine cmd) {
     public PicocliCommands(Supplier<Path> workDir, CommandLine cmd) {
         this.workDir = workDir;
         this.cmd = cmd;
-        commands = cmd.getCommandSpec().subcommands().keySet();
+        commands = new HashSet<>(cmd.getCommandSpec().subcommands().keySet());
         for (String c: commands) {
             for (String a: cmd.getSubcommands().get(c).getCommandSpec().aliases()) {
                 aliasCommand.put(a, c);
             }
         }
     }
 
+    public void includeClearScreenCommand(LineReader reader) {
+    	if (reader == null) return;
+    	ClearScreen clearScreen = new ClearScreen((LineReaderImpl) reader);
+    	cmd.addSubcommand(clearScreen);
+
+    	commands.add("clear");
+
+    	aliasCommand.put("clear", "clear");
+    	aliasCommand.put("cls", "clear");
+    }
+
```

Comment:
It is a bit unfortunate that we cannot do this declaratively, like `@Command(subcommands = { (...), ClearScreen.class, (...)}` in current master.

Is this necessary because we need to set the `LineReaderImpl`, or are there other reasons as well?

See also my suggestion below for installing the ClearScreen command as an alternative to `picocliCommands.includeClearScreenCommand(reader);`

Reply:
Please see the comment below.